server/config/passport.js

```diff
@@ -0,0 +1,57 @@
+const bcrypt = require("bcrypt");
+const passport = require("passport");
+const JwtStrategy = require("passport-jwt").Strategy;
+const LocalStrategy = require("passport-local").Strategy;
+const User = require("../models/User");
+
+module.exports = passport.use(
+  new LocalStrategy(async (username, password, done) => {
+    try {
+      const user = await User.findOne({ username });
+      if (!user) {
+        return done(null, false, {
+          message: "Username does not exist",
```

Comment:
For login API, you should always return the same error message (Incorrect username and password), it's a security flaw you can read about online